### ISSUE TITLE
Product Categories List count values are not accessible

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#1674]
**To Reproduce**
1. Create a post with a _Product Categories List_ block.
2. With a screen reader navigate it.
3. Notice the number is read without any unit.

**Expected behavior**
Ideally it should read `14 products` (instead of just `14`).

**Screenshots**
![imatge](https://user-images.githubusercontent.com/3616980/73384605-dc4c3e80-42cb-11ea-8c59-3eb93d65aa71.png)

---

This issue affects the list and the dropdown views.

For inspiration, something similar is solved in the [_Attributes Filter_ block](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/blocks/attribute-filter/label.js).
